src/pyscript/plugins/create.py

```diff
@@ -20,25 +27,111 @@ def create(
         "--project-type",
         help="Type of project that is being created. Supported types are: 'app' or 'plugin'",
     ),
+    wrap: Optional[bool] = typer.Option(
+        False,
+        "-w",
+        "--wrap",
+        help="Use wrap mode i.e. embed a python script into an HTML file",
+    ),
+    output: Optional[Path] = typer.Option(
+        None,
+        "-o",
+        "--output",
+        help="""Path to the resulting HTML output file. Defaults to input_file with suffix replaced.
+        Meant to be used with `--wrap`""",
+    ),
+    command: Optional[str] = typer.Option(
+        None,
+        "-c",
+        "--command",
+        help="If provided, embed a single command string. Meant to be used with `--wrap`",
+    ),
+    title: Optional[str] = typer.Option(
+        None, help="Add title to HTML file. Meant to be used with `--wrap`"
+    ),
+    show: Optional[bool] = typer.Option(
+        False,
+        "--show",
+        help="Open output file in web browser. Meant to be used with `--wrap`",
+    ),
 ):
     """
     Create a new pyscript project with the passed in name, creating a new
-    directory in the current directory.
+    directory in the current directory. Alternatively, use `--wrap` so as to embed
+    a python file instead.
     """
-    try:
-        create_project(
-            app_name,
-            app_description,
-            author_name,
-            author_email,
-            pyscript_version,
-            project_type,
+    if not app_or_file_name and not command:
+        raise cli.Abort(
+            "Must provide either an input '.py' file or a command with the '-c' option."
         )
-    except FileExistsError:
+
+    if app_or_file_name and command:
+        raise cli.Abort("Cannot provide both an input '.py' file and '-c' option.")
+
+    if (title or output or show or command) and (not wrap):
         raise cli.Abort(
-            f"A directory called {app_name} already exists in this location."
+            """`--title`, `--output/-o`, `--show` and `--command/-c`
+            are meant to be used with `--wrap/-w`"""
         )
 
+    if not wrap:
+        try:
+            if not app_description:
+                app_description = typer.prompt("App description")
+            if not author_name:
+                author_name = typer.prompt("Author name")
+            if not author_email:
+                author_email = typer.prompt("Author email")
+            create_project(
+                str(app_or_file_name),
```

Comment:
If I am reading the code correct `app_or_file_name` may still be None here right? If that's the case we probably don't want to str a None value 🤔

Reply:
It won't matter since there are checks that prevent this + if you don't pass anything i.e. if it is None, then `typer` has safeguards aka it will prompt that you need to enter an argument

Reply:
Do you mean that `create_project` checs if the `app_or_file_name` is the string `None` and remove it? Because that's what will happen correct?

Reply:
No, not the `create_project` function, but the `pyscript create` command, that uses typer.

Reply:
I might be missing something, we set `app_or_file_name` with:

```
app_or_file_name: Optional[Path] = typer.Argument(
        None, help="The name of your new app or path to an input .py script"
    ),
```

So that means this could be none since that's the default? Because we then have:

```
if not app_or_file_name and not command:
```

Which makes me assume that app_or_file_name can indeed be none? 🤔

Reply:
Ideally, typer was handling for us because we used `typer.Argument` instead of `typer.Option`

But, I did come across a case when we use `-w` and `-c` together i.e. `wrap` and `command`.

In this case, indeed `app_or_file_name` is kinda `None`, I have handled it now.